File: src/main/java/plugin/object/WildernessDitchObjectPlugin.java
package plugin.object;

import org.redrune.engine.tick.task.WorldTask;
import org.redrune.engine.tick.task.WorldTasksManager;
import org.redrune.game.content.plugin.type.ObjectPlugin;
import org.redrune.game.entity.actor.mask.Animation;
import org.redrune.game.entity.actor.mask.ForceMovement;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.entity.object.WorldObject;
import org.redrune.game.global.WorldTile;
import org.redrune.utility.game.ClickOption;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-01-31
 */
public class WildernessDitchObjectPlugin implements ObjectPlugin {
	
	@Override
	public boolean handle(Player player, WorldObject object, String option) {
		performJump(player, object, null);
		return true;
	}
	
	@Override
	public void register() {
		for (int i = 1440; i <= 1444; i++) {
			registerSpecifiedOption(ClickOption.FIRST, i);
		}
	}
	
	public static void performJump(Player player, WorldObject object, final Runnable onJump) {
		player.getLocks().lock();
		player.setNextAnimation(new Animation(6132));
		final boolean in = player.getY() < object.getY();
		final WorldTile toTile = new WorldTile(player.getX(), in ? object.getY() + 2 : object.getY() - 1, object.getPlane());
		player.setNextForceMovement(new ForceMovement(new WorldTile(player), 1, toTile, 2, in ? ForceMovement.NORTH : ForceMovement.SOUTH));
		WorldTasksManager.schedule(new WorldTask() {
			
			@Override
			public void run() {
				WorldTile faceTile = new WorldTile(player.getX(), player.getY() + (in ? 1 : -1), player.getPlane());
				player.setNextFaceWorldTile(faceTile);
				player.setNextWorldTile(toTile);
				player.getLocks().unlock();
				if (onJump != null) {
					onJump.run();
				}
			}
		}, 2);
	}
}

File: src/main/java/plugin/object/WildernessDitchObjectPlugin.kt
package plugin.`object`

import engine.tick.task.WorldTask
import engine.tick.task.WorldTasksManager
import game.content.plugin.type.ObjectPlugin
import game.entity.`object`.WorldObject
import game.entity.actor.mask.Animation
import game.entity.actor.mask.ForceMovement
import game.entity.actor.player.Player
import game.global.WorldTile
import utility.game.ClickOption

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-01-31
 */
class WildernessDitchObjectPlugin : ObjectPlugin {
    override fun handle(player: Player, `object`: WorldObject, option: String): Boolean {
        performJump(player, `object`, null)
        return true
    }

    override fun register() {
        for (i in 1440..1444) {
            registerSpecifiedOption(ClickOption.FIRST, i)
        }
    }

    companion object {
        @JvmStatic
        fun performJump(player: Player, `object`: WorldObject, onJump: Runnable?) {
            player.locks.lock()
            player.nextAnimation = Animation(6132)
            val `in`: Boolean = player.y < `object`.y
            val toTile =
                WorldTile(player.x, if (`in`) `object`.y + 2 else `object`.y - 1, `object`.plane)
            player.nextForceMovement =
                ForceMovement(WorldTile(player), 1, toTile, 2, if (`in`) ForceMovement.NORTH else ForceMovement.SOUTH)
            WorldTasksManager.schedule(object : WorldTask() {
                override fun run() {
                    val faceTile = WorldTile(player.x, player.y + if (`in`) 1 else -1, player.plane)
                    player.nextFaceWorldTile = faceTile
                    player.setNextWorldTile(toTile)
                    player.locks.unlock()
                    onJump?.run()
                }
            }, 2)
        }
    }
}